### PR TITLE
Feature/tdkn-85 : jsonschema no java class serialization

### PR DESCRIPTION
**Pull request by sgandon**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [x] Feature

**What is the current behavior?** (You can also link to an open issue here)
The Properties serialization was adding the instance class name in the json schema string.

**What is the new behavior?**
it does not anymore.
But the instance of the Properties is required by the de-serializer.

**Does this PR introduce a breaking change?**

- [x] Yes

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...
You need to create and instance of the Properties before calling
`org.talend.daikon.serialize.jsonschema.JsonSchemaUtil.fromJson(InputStream, Properties)
`
Also this utility class has been renamed from *JsonUtil* to *JsonSchemaUtil*

### CODE REVIEW

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/131/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/131/)

**Comment by sgandon** [comment]
I also commited unrelated changes to this PR such as the deprecation of 2 RuntimeException method to use more useful ones that get correctly parsed in the IDEs.
And made the Form constructor private cause it is only used for deserialization and should never be used but developers.

File: daikon/src/main/java/org/talend/daikon/exception/TalendRuntimeException.java [hunk 1]
```diff
@@ -103,18 +103,38 @@ public TalendRuntimeException(ErrorCode code) {
...
         throw new TalendRuntimeException(CommonErrorCodes.UNEXPECTED_EXCEPTION, cause);
     }
 
+    /**
+     * Called when something unexpected happens.
+     * 
+     * @param cause the unexpected exception.
+     * @deprecated cause the IDEs do not know this throws an exception please use {@link #createUnexpectedException(String)}
+     */
+    @Deprecated
```
Comment (RyanSkraba) [inline daikon/src/main/java/org/talend/daikon/exception/TalendRuntimeException.java]: You can probably just get rid of this, nobody uses it but daikon!

File: daikon/src/main/java/org/talend/daikon/properties/PropertiesImpl.java [hunk 2]
```diff
@@ -604,4 +608,41 @@ public String toStringIndent(int indent) {
...
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
+            return false;
+        }
+        return this.hashCode() == obj.hashCode();
```
Comment (RyanSkraba) [inline daikon/src/main/java/org/talend/daikon/properties/PropertiesImpl.java]: ... two non-equal objects can have the same hashCode! :(

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/134/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/134/)

File: daikon/src/main/java/org/talend/daikon/exception/TalendRuntimeException.java (hunk 1 above)
Comment (sgandon) [inline daikon/src/main/java/org/talend/daikon/exception/TalendRuntimeException.java]: This may be used by the Studio.

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/136/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/136/)

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/137/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/137/)

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/139/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/139/)

**Pull request merged**